### PR TITLE
feat(integration-directory): update project select modal

### PR DESCRIPTION
**Pull request by scefali**
This PR changes the project/org select modal to use the design from Chris without the changes to the modal header. As a team, we decided that we should tackle the modal header change globally and not change this particular one in isolation. The main stylistic change is the border of the items in the dropdown.

I made the text dynamic based on whether the user has to pick a project and/or an organization. I've also removed the `deprecatedSelectControl` prop so I can apply custom styling to the dropdown.

Before:
Screen Shot 2020-02-26 at 2 39 26 PM

After:
Screen Shot 2020-02-26 at 3 52 38 PM

Please note I also had to revert the changes in an earlier PR in order to upgrade React Select in this component (https://github.com/getsentry/sentry/pull/16439). This PR would pre-focus the select the project when the user was on an issue page for that project.

The reason for this is because the function of `focusOption` has been deprecated. There is no way to focus on an option other than by changing the current value of the select. That would require making this a controlled component and change the value to the current project once the projects load. This caused a bug the first time I made this PR without this change: https://sentry.io/organizations/sentry/issues/1536012493/?project=11276&referrer=slack

While it can beneficial to pre-select the project, the use case for this is narrow. It only happens when you have a projectId as query parameter (e.g. issue page) and not on a settings page for a project because in that case, it will simply skip the project selection step and just go to that project. As far as I can tell, it can only happen in two scenarios: when the user presses `command + k` to open the search or during onboarding to set up releases. And the user must be on an issue page in both cases. If it's during onboarding, the user should have so few projects as to make pre-selecting the project unimportant. And I don't think many users even know about opening the search bar with `command + k` (I only learned of it today). Therefore, I believe that rolling back this feature will have a negligible impact on users.

### CODE REVIEW

**Comment by matejminar** [comment]
@jan-auer @mitsuhiko do you have any strong feelings about this? I think that you two are heavy users of this feature.

File: src/sentry/static/sentry/app/components/contextPickerModal.tsx [hunk 1]
```diff
@@ -228,15 +222,24 @@ class ContextPickerModal extends React.Component<Props> {
       .filter(({status}) => status.id !== 'pending_deletion')
       .map(({slug}) => ({label: slug, value: slug}));
 
+    const customStyles = {
+      option: provided => ({
+        ...provided,
+        borderTop: `1px solid ${theme.gray1}`,
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/contextPickerModal.tsx]: Why the one off style here?

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/contextPickerModal.tsx]: If this is also to match @ckj's design, we should maybe implement this in other select boxes?

I think visually this is a little wonky too

File: src/sentry/static/sentry/app/components/contextPickerModal.tsx [hunk 2]
```diff
@@ -203,6 +183,21 @@ class ContextPickerModal extends React.Component<Props> {
...
+  get headerText() {
+    const {needOrg, needProject} = this.props;
+    if (needOrg && needProject) {
+      return t('Select an organization and a project to continue');
+    }
+    if (needOrg) {
+      return t('Select an organization to continue');
+    }
+    if (needProject) {
+      return t('Select an a project to continue');
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/contextPickerModal.tsx]:
```suggestion
      return t('Select a project to continue');
```

File: src/sentry/static/sentry/app/components/contextPickerModal.tsx [hunk 3]
```diff
@@ -228,15 +222,24 @@ class ContextPickerModal extends React.Component<Props> {
...
       .map(({slug}) => ({label: slug, value: slug}));
 
+    const customStyles = {
+      option: provided => ({
+        ...provided,
+        borderTop: `1px solid ${theme.gray1}`,
+      }),
+      menuList: provided => ({
+        ...provided,
+        paddingTop: '0px',
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/contextPickerModal.tsx]:
```suggestion
        paddingTop: 0,
```

I believe we don't need to specify units when the value is 0

**Comment by billyvg** [comment]
Why would we focus the option instead of just selecting it outright?

**Comment by scefali** [comment]
@billyvg I can do that if I make it a controlled component. I’ll go with that approach.

**Comment by billyvg** [comment]
@scefali I'm not necessarily suggesting that's the best solution, was trying to find out why `focusOption` in the first place.

**Comment by matejminar** [comment]
@scefali @billyvg 
Motivation is simple - the command palette should be optimized for speed and convenience.

When you want to go for example from issue detail page to Client keys you type DSN, hit Enter and then Enter to confirm project. 

image

There is no reason for us to not prefocus the current project when we have that information - the user would otherwise need to realize what project he is in, type it in and then hit Enter.

This was a big pain point for some folks in Vienna that use this feature daily.

We can't preselect instead of prefocus because of current buttonless design that takes you to next step upon selection (and sometimes you want to go to other than current project)

image

**Pull request closed without merging**